Library/Homebrew/os/linux/elf.rb

```diff
@@ -68,28 +68,26 @@ def binary_executable?
     elf_type == :executable
   end
 
-  def with_interpreter?
-    return @with_interpreter if defined? @with_interpreter
+  def interpreter
+    return @interpreter if defined? @interpreter
 
-    @with_interpreter = if binary_executable?
-      true
-    elsif dylib?
-      if HOMEBREW_PATCHELF_RB
-        begin
-          patchelf_patcher.interpreter.present?
-        rescue PatchELF::PatchError => e
-          opoo e unless e.to_s.start_with? "No interpreter found"
-          false
-        end
-      elsif which "readelf"
-        Utils.popen_read("readelf", "-l", to_path).include?(" INTERP ")
-      elsif which "file"
-        Utils.popen_read("file", "-L", "-b", to_path).include?(" interpreter ")
-      else
-        raise "Please install either readelf (from binutils) or file."
+    if HOMEBREW_PATCHELF_RB
```

Comment:
```suggestion
    @interpreter ||= if HOMEBREW_PATCHELF_RB
```

Reply:
My previous `||=` suggestion was wrong because `nil` is falsey.
```suggestion
    @interpreter = if HOMEBREW_PATCHELF_RB
```